app/src/mainwindow2.cpp

```diff
@@ -416,6 +421,68 @@ void MainWindow2::clearRecentFilesList()
     getPrefDialog()->updateRecentListBtn(!recentFilesList.isEmpty());
 }
 
+void MainWindow2::pegBarReg()
+{
+    if (mEditor->layers()->currentLayer()->type() != Layer::BITMAP)
+    {
+        QMessageBox::information(this, nullptr,
+                                 tr("Layer type not supported!"),
+                                 QMessageBox::Ok);
+        return;
+    }
+
+    if (!ui->scribbleArea->isSomethingSelected())
+    {
+        QMessageBox::information(this, nullptr,
+                                 tr("Please select an area around center peg!\n"
+                                    "Area must cover all pegs to be aligned."),
+                                 QMessageBox::Ok);
+        return;
+    }
+
+    if (!mEditor->layers()->currentLayer()->keyExists(mEditor->currentFrame()))
+    {
+        QMessageBox::information(this, nullptr,
+                                 tr("No reference Key selected!"),
+                                 QMessageBox::Ok);
+        return;
+    }
+
+    PegBarRegistration* pegreg = new PegBarRegistration(this);
+    QStringList bitmaplayers;
+    for (int i = 0; i < mEditor->layers()->count(); i++)
```

Comment:
I had it inside the class, but I use LayerManager to sort out which layers are Bitmap layers. Since core files are not allowed in app, I had to move it to MainWindow2.